### PR TITLE
gpgme: 1.23.2 -> 1.24.0

### PR DESCRIPTION
**Pull request by dotlambda**
fixes https://github.com/NixOS/nixpkgs/issues/354166
cc @lucc

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/gpgme/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "gpgme";
+  inherit (gpgme) version src patches;
+  pyproject = true;
+
+  postPatch = ''
+    substituteInPlace lang/python/setup.py.in \
+      --replace "gpgme_h = '''" "gpgme_h = '${lib.getDev gpgme}/include/gpgme.h'"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/gpgme/default.nix:17]: This should be replace-fail

File: pkgs/development/python-modules/gpgme/default.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  buildInputs = [
+    gpgme
+  ];
+
+  pythonImportsCheck = [ "gpg" ];
+
+  meta = {
+    description = "Python bindings to the GPGME API of the GnuPG cryptography library";
+    homepage = "https://dev.gnupg.org/source/gpgme/browse/master/lang/python/";
+    inherit (gpgme.meta) changelog license maintainers;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/gpgme/default.nix:47]: We should merge all of meta to not forget anything, right?

Comment (dotlambda) [reply pkgs/development/python-modules/gpgme/default.nix:47]: I thought that creates problems with `meta.position` but it must be some other problem.

Comment (dotlambda) [reply pkgs/development/python-modules/gpgme/default.nix]: @eclairevoyant Can you explain why I get
```
nix-repl> python3.pkgs.gpgme.meta.position
"nixpkgs/pkgs/development/python-modules/gpgme/default.nix:45"
```
here despite using `meta = gpgme.meta // ...`? C.f. https://github.com/NixOS/nixpkgs/pull/324127#discussion_r1712991684

Comment (Atemu) [reply pkgs/development/python-modules/gpgme/default.nix]: It's because, not despite.

Comment (dotlambda) [reply pkgs/development/python-modules/gpgme/default.nix]: But removing `gpgme.meta //` yields the same `meta.position`.

File: pkgs/development/python-modules/gpgme/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
+{
+  automake,
```
Comment (lucc) [inline pkgs/development/python-modules/gpgme/default.nix:2]: unused?

File: pkgs/development/python-modules/gpgme/default.nix [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  buildPythonPackage,
+  gpgme,
+  lib,
+  setuptools,
+  swig,
+}:
+
+buildPythonPackage rec {
```
Comment (lucc) [inline pkgs/development/python-modules/gpgme/default.nix:10]: `rec` unused?

File: pkgs/development/python-modules/gpgme/default.nix [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+  nativeBuildInputs = [
+    swig
+  ];
+
+  buildInputs = [
+    gpgme
+  ];
+
+  pythonImportsCheck = [ "gpg" ];
```
Comment (lucc) [inline pkgs/development/python-modules/gpgme/default.nix:42]: :+1: that would have saved me in #354166   :)

**Review by lucc — COMMENTED** [review]
Just some code nitpicks. Thanks for splitting it into a proper python derivation with the import check.

Do I understand the code correctly, that the python derivation depends on the binary library derivation and also executes the configure script in the sources again but does not build the binary library again?

**Comment by dotlambda** [comment]
> Do I understand the code correctly, that the python derivation depends on the binary library derivation and also executes the configure script in the sources again but does not build the binary library again?

Yes. The configure script at least sets the version, i.e. replaces `@version@`, while turning `setup.py.in` into `setup.py`. I could do that manually instead but I don't think there's a lot of harm in running the script.

**Comment by dotlambda** [comment]
The question is: Is this acceptable for staging-next? Otherwise it won't land in 24.11.
I tried only applying the second commit but ran into problems, even after adding `distutils` to `build-inputs`.
cc @NixOS/nixos-release-managers

**Comment by RossComputerGuy** [comment]
> The question is: Is this acceptable for staging-next? 

I'm not sure, lemme ask the staging folks.

> Otherwise it won't land in 24.11.

Yeah, we should fix this before 24.11's release.

**Comment by RossComputerGuy** [comment]
Well it looks like this won't make it to 24.11, should go back to staging and it can land in there once approved enough.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by dotlambda** [comment]
I don't understand why `python313.pkgs.gpgme` fails to build with
```
checking for swig... /nix/store/8w5lfkjysan93sv31rmjxbpw1g1iv115-swig-4.2.1/bin/swig
checking for a Python interpreter with version >= 2.7... none
checking for a Python interpreter with version >= 3.6... none
checking for a Python interpreter with version >= 3.8... none
checking for a Python interpreter with version >= 3.9... none
checking for a Python interpreter with version >= 3.10... none
checking for a Python interpreter with version >= 3.11... none
checking for a Python interpreter with version >= 3.12... none
checking for a Python interpreter with version >= all... none
configure: error: 
***
*** Please install the python development packages.
***
```
@mweinelt Any idea?

**Comment by lucc** [comment]
@dotlambda I think that is the configure script of gpgme which tries to find a suitable python version and they have hardcoded a list :/

Maybe there is an option to tell configure to just use one specific python executable and not search for  it, but I did not find that option at first glance.

**Comment by dotlambda** [comment]
Should we merge this anyway and fix Python 3.13 later?

**Comment by dotlambda** [comment]
@ofborg build gpgme.passthru.tests python313.pkgs.gpgme

**Comment by FliegendeWurst** [comment]
Fix for python313Packages.gpgme: #371188

**Pull request merged**